docs/design/permissive-mode.md

```diff
@@ -0,0 +1,353 @@
+# Hermeto Permissive Mode in SBOM
+
+## Overview
+
+We have added a CLI option `--mode` to allow users to have some control over the
+strictness of the validation checks. It is set to `strict` by default. The other
+option is `permissive`.
+
+This document describes the design for tracking permissive mode violations during
+SBOM generation in Hermeto. The goal is to provide granular tracking of these
+violations in the SBOM.
+
+## Context
+
+Hermeto, in permissive mode, bypasses certain validation checks and continues
+processing even when issues are encountered. However, this information is not
+currently captured in the generated SBOMs, making it difficult to audit what
+violations occurred during processing.
+
+The strict mode remains as the default, and the permissive mode is reserved for
+rare and deliberately audited exceptions. By embedding detailed information into
+the SBOM, we will ensure that a more granular and precise policy can be
+implemented in the future.
+
+Validation checks that are currently relaxed in the permissive mode:
+
+- `gomod` - vendor directory contents differ from go.mod/go.sum after vendoring
+- `cargo` - Cargo.lock is out of sync with Cargo.toml (it must exist though)
+
+## Goals
+
+- Record specific violation, not the whole permissive mode
+- Compatibility with both supported SBOM formats - CycloneDX and SPDX
+- Machine-readable format for easy parsing and processing
+
+## Solution
```

Comment:
Non-blocking nitpick: s/Solution/Proposed Solutions/ because there are many solutions and some of them are not selected.